### PR TITLE
Don't build unless the branch is master

### PR DESCRIPTION
**Pull request by tlunter**

### CODE REVIEW

**Comment by tlunter** [comment]
@adamjt Think this should do better

**Comment by tlunter** [comment]
I have no idea how to build this thing!

File: src/main/java/hudson/plugins/git/GitSCM.java [hunk 1]
```diff
@@ -648,8 +648,10 @@ private PollingResult compareRemoteRevisionWithImpl(AbstractProject<?, ?> projec
             listener.getLogger().println("[poll] Last Built Revision: " + buildData.lastBuild.revision);
         }
 
-        final String singleBranch = "master";
-        // final String singleBranch = getSingleBranch(lastBuild);
+        final String singleBranch = getSingleBranch(lastBuild);
+        if (!singleBranch.equals("master")) {
+                return PollingResult.NO_CHANGES;
```
Comment (adamjt) [inline src/main/java/hudson/plugins/git/GitSCM.java:653]: Will this return no changes even if the build was triggered by a push to master?

Comment (tlunter) [inline src/main/java/hudson/plugins/git/GitSCM.java:653]: Good point, will check.

**Comment by tlunter** [comment]
Looks like there is waaaay too much going on in just determining this branch so I'm just gonna close this.

**Pull request closed without merging**